instrumentation/apolloconfig-apolloclient-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apolloconfig/apolloclient/v1_0/ApolloConfigInstrumentationModule.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apolloconfig.apolloclient.v1_0;
+
+import static java.util.Collections.singletonList;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import java.util.List;
+
+@AutoService(InstrumentationModule.class)
+public class ApolloConfigInstrumentationModule extends InstrumentationModule {
+  public ApolloConfigInstrumentationModule() {
```

Comment:
I think this instrumentation should be disabled by default as the telemetry it produces is not described in the otel semantic conventions in any way.

Reply:
This is worth discussing, and I think this problem is general.

For example, in the scenario of using MQ, it may not be correct, but my current approach is to add a span of the Consumer type when consuming, and link it to the Producer span. I think this is more elegant.

However, in the configuration change scenario, we seem to be unable to add the Producer span to all trigger sources

For example, in the well-known k8s scenario, we watch the changes in CRD and, assuming that there are many remote call to api-server in our Reconcile logic , I think adding a parent span is very helpful for locating the problem.

Reply:
> This is worth discussing, and I think this problem is general.

We prefer to set the instrumentation to be disabled default, if the telemetry it produces is not described in any OTel semantic conventions. Just like we'd done in MyBatis: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/7b69e56cb632c4dd08c99ab3d3ba01a603cd236d/instrumentation/mybatis-3.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/mybatis/v3_2/MyBatisInstrumentationModule.java#L29-L31

Reply:
OK, I will set it to be false by default

After I removed testLatestDeps, CI test-latest-deps / testLatestDeps1 reported an error. I was overwhelmed by the logs. Can you help me check it out?

Reply:
> After I removed testLatestDeps, CI test-latest-deps / testLatestDeps1 reported an error. I was overwhelmed by the logs. Can you help me check it out?

See https://scans.gradle.com/s/wb7ejo6uepn2w/tests/task/:instrumentation:apolloconfig-apolloclient-1.0:javaagent:test/details/io.opentelemetry.javaagent.instrumentation.apolloconfig.apolloclient.v1_0.ApolloRepositoryChangeTest?expanded-stacktrace=WyIwIl0&top-execution=1
and read https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12794#discussion_r1867496874 for instructions on how to fix it.